symengine/dense_matrix.cpp

```diff
@@ -1520,6 +1520,45 @@ void inverse_gauss_jordan(const DenseMatrix &A, DenseMatrix &B)
     fraction_free_gauss_jordan_solve(A, e, B);
 }
 
+// ----------------------- Vector-specific Methods --------------------------//
+
+void dot(const DenseMatrix &A, const DenseMatrix &B, DenseMatrix &C)
+{
+    if (A.col_ == B.row_) {
+        if (B.col_ != 1) {
+            DenseMatrix tmp1 = DenseMatrix(A.col_, A.row_);
+            A.transpose(tmp1);
+            DenseMatrix tmp2 = DenseMatrix(B.col_, B.row_);
+            B.transpose(tmp2);
+            C.resize(tmp1.row_, tmp2.col_);
+            mul_dense_dense(tmp1, tmp2, C);
+        } else {
+            C.resize(A.row_, B.col_);
+            mul_dense_dense(A, B, C);
+        }
+    } else if (A.col_ == B.col_) {
+        DenseMatrix tmp2 = DenseMatrix(B.col_, B.row_);
+        B.transpose(tmp2);
+        dot(A, tmp2, C);
+    } else if (A.row_ == B.row_) {
+        DenseMatrix tmp1 = DenseMatrix(A.col_, A.row_);
+        A.transpose(tmp1);
+        dot(tmp1, B, C);
```

Comment:
I think that this case is redundant, even though `SymPy` has  it implemented. I'll remove it.